### PR TITLE
Prevent instantiation of top_metrics when sub-aggregations are present

### LINKED ISSUE #95663: top_metrics aggregation should fail if it contains sub aggregations
top_metrics is a metric aggregation that does not support sub aggregations. Currently if you add sub aggregation they will be silently ignored. Instead it should throw an error reporting it to the user. It is very easy to reproduce by using the example in the docs:

```
POST /test/_bulk?refresh
{"index": {}}
{"s": 1, "m": 3.1415}
{"index": {}}
{"s": 2, "m": 1.0}
{"index": {}}
{"s": 3, "m": 2.71828}

POST /test/_search?filter_path=aggregations
{
  "aggs": {
    "tm": {
      "top_metrics": {
        "metrics": {"field": "m"},
        "sort": {"s": "desc"}
      },
      "aggs": {
        "min": {
          "min": {
            "field": "m"
          }
        }
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by shans96**
Took inspiration from some of the other inheritors of `AbstractAggregationBuilder` to fix this bug, although I'm not sure if I need to override any of the other `subAggregation` methods inside the top metrics builder to completely fix the issue.

I was also wondering if there's a way to make use of the existing `createTestInstance` method to reduce the size of the new test, but it doesn't seem like I can use it because I need to set the aggregator name in advance.

Closes #95663

### CODE REVIEW

File: x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregationBuilder.java [hunk 1]
```diff
@@ -169,6 +170,20 @@ protected void doWriteTo(StreamOutput out) throws IOException {
...
+    @Override
+    public TopMetricsAggregationBuilder subAggregations(Builder subFactories) {
+        throw new AggregationInitializationException(
+            "Aggregator [" + name + "] of type [" + getType() + "] cannot accept sub-aggregations"
+        );
+    }
+
+    @Override
+    public TopMetricsAggregationBuilder subAggregation(AggregationBuilder aggregation) {
+        throw new AggregationInitializationException(
```
Comment (iverase) [inline x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregationBuilder.java:182]: I think this is not needed and can be removed.

**Review by iverase — CHANGES_REQUESTED** [review]
Thanks! could you please remove the overrided method #subAggregation(AggregationBuilder aggregation) as it is not needed. After that we can merge the change.

File: x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregationBuilder.java (hunk 1 above)
Comment (shans96) [reply x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregationBuilder.java]: Thanks, updated

**Comment by iverase** [comment]
@elasticmachine please test this

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by shans96** [comment]
Forgot to update test, fixing now

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by shans96** [comment]
Currently fixing an issue with my local build so I can run `./gradlew check` , but for now I've been relying on CI output, sorry for any inconvenience. For next time, I wanted to know if it's fine to tag elasticmachine myself, once ready?

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by iverase** [comment]
> For next time, I wanted to know if it's fine to tag elasticmachine myself if I miss something small?

I think you cannot tag the bot, it is too clever.

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine test this please

**Review by iverase — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
Thank you for the contribution @shans96!

**Pull request merged**